src/hypothesis_jsonschema/_canonicalise.py

```diff
@@ -665,6 +665,22 @@ def diff_keys(k: str) -> bool:
         ):
             out[key] = max([out[key], s.pop(key)])
         # TODO: Handle remaining mergable keys.
+        if "multipleOf" in out and "multipleOf" in s:
+            if type(out["multipleOf"]) == int and type(s["multipleOf"]) == int:
+                all_prime_factors = prime_fac(out["multipleOf"]) + prime_fac(
+                    s["multipleOf"]
+                )
+                prime_product = 1
+                for x in all_prime_factors:
+                    prime_product *= x
+                union = []
+                for x in prime_fac(out["multipleOf"]):
+                    if x in prime_fac(s["multipleOf"]):
+                        union.append(x)
+                union_product = 1
+                for x in union:
+                    union_product *= x
+                out["multipleOf"] = prime_product // union_product
```

Comment:
```suggestion
        if isinstance(out.get("multipleOf"), int) and isinstance(s.get("multipleOf"), int):
            x, y = s.pop("multipleOf"), out["multipleOf"]
            out["multipleOf"] = x * y // gcd(x, y)
```

And then we don't need prime factors.

Reply:
Thanks! I do love solving things with prime numbers, but your solution is much more concise!